hail/python/hailtop/utils/gcs_requester_pays.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any, DefaultDict, Dict, FrozenSet, Optional, Tuple, Type, TypeVar, Union
+
+from hailtop.aiocloud.aiogoogle import GCSRequesterPaysConfiguration
+
+
+FS = TypeVar("FS")
+MaybeGCSRequesterPaysConfiguration = Optional[GCSRequesterPaysConfiguration]
+GCSRequesterPaysKey = Optional[Union[str, Tuple[str, FrozenSet[str]]]]
+
+
+class GCSRequesterPaysFSCache(DefaultDict[MaybeGCSRequesterPaysConfiguration, FS]):
+    def __init__(self, *args: Any, fs_constructor: Type[FS], default_kwargs: Dict[str, Any] = {}, **kwargs: Any) -> None:
+        self._fs_constructor = fs_constructor
+        self[None] = fs_constructor(**default_kwargs)
```

Comment:
I feel like a cache shouldn't eagerly construct elements. Can we delete this and let the normal behavior deal with this case?

Reply:
updated!